File: dotnet/src/xjustiz.core-dotnet/Models/XjustizFachspezifischeDaten.cs
namespace xjustiz.core_dotnet.Models;

using System.Xml.Serialization;
using xjustiz.core_dotnet.Models.Codes;
using xjustiz.core_dotnet.Models.Entities;
using xjustiz.core_dotnet.Util.Versioning;

/// <summary>
/// Fachspezifische Daten eines Dokuments.
/// Specialized data of a document.
/// </summary>
[XJustizAvailability(XJustizVersion.V3_1_1)]
[XJustizCoreAvailability(XJustizCoreVersion.V1_0_0)]
public class XjustizFachspezifischeDaten
{
    /// <summary>
    /// Bezeichnet die Dokumentklasse anhand der vorgegebenen Werteliste. Diese ist für einen reibungslosen, systemübergreifenden Austausch bewusst knappgehalten. Für eine spezifischere Bezeichnung kann das Element 'dokumententyp' verwendet werden.
    /// Designates the document class based on the predefined value list. This is deliberately kept concise for smooth, cross-system exchange. For a more specific designation, the element 'dokumententyp' (document type) can be used.
    /// </summary>
    [XmlElement("dokumentklasse", Namespace = XJustizExport.Tns, Order = 1)]
    [XJustizAvailability(XJustizVersion.V3_1_1)]
    [XJustizCoreAvailability(XJustizCoreVersion.V1_0_0)]
    public DokumentklasseCode? Dokumentklasse { get; set; }

    /// <summary>
    /// Repräsentiert eine Datei.
    /// Represents a file.
    /// </summary>
    [XmlElement("datei", Namespace = XJustizExport.Tns, Order = 2)]
    [XJustizAvailability(XJustizVersion.V3_1_1)]
    [XJustizCoreAvailability(XJustizCoreVersion.V1_0_0)]
    public Datei Datei { get; set; } = new();
}

File: java/src/main/java/de/xjustiz/core/models/XjustizFachspezifischeDaten.java
package de.xjustiz.core.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import jakarta.xml.bind.annotation.XmlAccessType;
import jakarta.xml.bind.annotation.XmlAccessorType;
import jakarta.xml.bind.annotation.XmlElement;
import org.jetbrains.annotations.Nullable;

/**
 * Fachspezifische Daten eines Dokuments.
 * <p>
 * <u><b>Subject-specific data:</b></u> Specialized data of a document.
 */
@XmlAccessorType(XmlAccessType.FIELD)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class XjustizFachspezifischeDaten {

    /**
     * Bezeichnet die Dokumentklasse anhand der vorgegebenen Werteliste. Diese ist
     * für einen reibungslosen, systemübergreifenden Austausch bewusst
     * knappgehalten. Für eine spezifischere Bezeichnung kann das Element
     * 'dokumententyp' verwendet werden.
     * <p>
     * <u><b>Document class:</b></u> Designates the document class based on the
     * predefined value list. This is deliberately kept concise for smooth,
     * cross-system exchange. For a more specific designation, the element
     * 'dokumententyp' (document type) can be used.
     */
    @XmlElement(name = "dokumentklasse", namespace = "http://www.xjustiz.de")
    @JacksonXmlProperty(localName = "dokumentklasse", namespace = "http://www.xjustiz.de")
    @JsonProperty("Dokumentklasse")
    @Nullable
    private Dokumentklasse dokumentklasse;

    /**
     * Repräsentiert eine Datei.
     * <p>
     * <u><b>File:</b></u> Represents a file.
     */
    @XmlElement(name = "datei", namespace = "http://www.xjustiz.de")
    @JacksonXmlProperty(localName = "datei", namespace = "http://www.xjustiz.de")
    @JsonProperty("Datei")
    private Datei datei;

    /**
     * Default constructor.
     */
    public XjustizFachspezifischeDaten() {
    }

    /**
     * Bezeichnet die Dokumentklasse anhand der vorgegebenen Werteliste. Diese ist
     * für einen reibungslosen, systemübergreifenden Austausch bewusst
     * knappgehalten. Für eine spezifischere Bezeichnung kann das Element
     * 'dokumententyp' verwendet werden.
     *
     * @return the dokumentklasse
     */
    @Nullable
    public Dokumentklasse getDokumentklasse() {
        return dokumentklasse;
    }

    /**
     * Sets the dokumentklasse.
     *
     * @param dokumentklasse the dokumentklasse to set
     */
    public void setDokumentklasse(@Nullable Dokumentklasse dokumentklasse) {
        this.dokumentklasse = dokumentklasse;
    }

    /**
     * Repräsentiert eine Datei.
     *
     * @return the datei
     */
    public Datei getDatei() {
        return datei;
    }

    /**
     * Sets the datei.
     *
     * @param datei the datei to set
     */
    public void setDatei(Datei datei) {
        this.datei = datei;
    }

}
